ql/time/date.cpp

```diff
@@ -826,6 +829,17 @@ namespace QuantLib {
         }
     }
 
+    std::size_t hash_value(const Date& d) {
+#ifdef QL_HIGH_RESOLUTION_DATE
+        std::size_t seed = 0;
+        boost::hash_combine(seed, d.serialNumber());
+        boost::hash_combine(seed, d.dateTime().total_nanoseconds());
```

Comment:
The `ptime` class no longer provides a `total_nanoseconds` method in recent versions of Boost.  I think this should work:
 ```suggestion
         boost::hash_combine(seed, d.dateTime().time_of_day().total_nanoseconds());
```